contracts/AllowedStrategy.sol

```diff
@@ -0,0 +1,270 @@
+// SPDX-License-Identifier: LGPL-3.0-only
+pragma solidity 0.8.11;
+
+import "@openzeppelin/contracts/utils/structs/EnumerableSet.sol";
+
+import "./common/UsingRegistryUpgradeable.sol";
+import "./common/UUPSOwnableUpgradeable.sol";
+import "./interfaces/IAccount.sol";
+import "./interfaces/IGroupHealth.sol";
+import "./interfaces/IManager.sol";
+import "./Managed.sol";
+
+contract AllowedStrategy is UUPSOwnableUpgradeable, UsingRegistryUpgradeable, Managed {
+    using EnumerableSet for EnumerableSet.AddressSet;
+
+    /**
+     * @notice Allowed strategies (validator groups) that can be chosen to voted on.
+     */
+    EnumerableSet.AddressSet private allowedStrategies;
+
+    /**
+     * @notice StCelo that was cast for allowed strategies
+     * strategy => stCelo amount
+     */
+    mapping(address => uint256) private allowedStrategyTotalStCeloVotes;
+
+    /**
+     * @notice Total StCelo that was voted with on allowed strategy
+     */
+    uint256 private totalStCeloInAllowedStrategies;
+
+    /**
+     * @notice Contract used for checking group health and rebalancing.
+     */
+    IGroupHealth public groupHealthContract;
+
+    /**
+     * @notice An instance of the Account contract this Manager manages.
+     */
+    IAccount internal account;
+
+    /**
+     * @notice Emitted when a new group is allowed for voting.
+     * @param group The group's address.
```

Comment:
Nit: should we keep open to other strategies being possible than just selecting a group? I.e. call this "strategy", more generally than just "group". Here and elsewhere.

Reply:
I updated it in Manager.sol